frontend/public/components/routes/create-route.tsx

```diff
@@ -0,0 +1,387 @@
+      .then(services => this.setState({
+        services,
+        loaded: true,
+      }))
+      .catch(err => this.setState({
+        error: err.message
+      }));
+  }
+
+  handleChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    const { name, value } = event.currentTarget;
+    this.setState({
+      [name]: value
+    } as any);
+  }
+
+  changeService = (serviceName: string) => {
+    const selectedService = _.find(this.state.services, { metadata: { name: serviceName } });
+    const portOptions = getPortOptions(selectedService);
+    const labels = selectedService.metadata.labels;
+    this.setState({
+      service: serviceName,
+      portOptions,
+      labels,
+    });
+  }
+
+  changeTargetPort = (portName: string) => this.setState({
+    targetPort: portName
+  });
+
+  toggleSection: React.ReactEventHandler<HTMLInputElement> = event => this.setState({
+    toggleData: {
+      ...this.state.toggleData,
+      [event.currentTarget.name]: event.currentTarget.checked
+    }
+  });
+
+  changeTerminationType = (terminationType: string) => {
+    const newState: any = {
+      terminationType,
+      // unset insecureTrafficType if it was set
+      insecureTrafficType: '',
+    };
+    switch (terminationType) {
+      case 'edge':
+        // unset tls data if it was set
+        newState.destinationCaCertificateFile = '';
+        break;
+      case 'passthrough':
+        Object.assign(newState, {
+          // unset tls data if it was set
+          certificateFile: '',
+          privateKeyFile: '',
+          caCertificateFile: '',
+          destinationCaCertificateFile: '',
+        });
+        break;
+      default:
+    }
+    this.setState(newState);
+  }
+
+  changeInsecureTrafficType = (insecureTrafficType: string) => this.setState({insecureTrafficType});
+
+  onCertificateFileChange = (certificateFile: string) => this.setState({certificateFile});
+
+  onPrivateKeyFileChange = (privateKeyFile: string) => this.setState({privateKeyFile});
+
+  onCaCertificateFileChange = (caCertificateFile: string) => this.setState({caCertificateFile});
+
+  onDestinationCaCertificateFileChange = (destinationCaCertificateFile: string) => this.setState({destinationCaCertificateFile});
+
+  save = event => {
+    event.preventDefault();
+
+    const {
+      name,
+      hostname,
+      path,
+      service,
+      targetPort,
+      terminationType,
+      insecureTrafficType,
+      certificateFile,
+      privateKeyFile,
+      caCertificateFile,
+      destinationCaCertificateFile,
+      toggleData: { secure },
+      namespace,
+      labels,
+    } = this.state;
+    const tls = secure
+      ? {
+        termination: terminationType,
+        insecureEdgeTerminationPolicy: insecureTrafficType,
+        certificate: certificateFile,
+        key: privateKeyFile,
+        caCertificate: caCertificateFile,
+        destinationCACertificate: destinationCaCertificateFile,
```

Comment:
nit: This becomes slightly easier if the state properties match the property names from the API.